src/splinterdb.c

```diff
@@ -785,3 +789,57 @@ splinterdb_close_print_stats(splinterdb *kvs)
    task_print_stats(kvs->task_sys);
    splinterdb_stats_print_insertion(kvs);
 }
+
+/*
+ * -------------------------------------------------------------------------
+ * External "APIs" provided mainly to invoke lower-level functions intended
+ * for use -ONLY- as testing interfaces.
+ * -------------------------------------------------------------------------
+ */
+void
+splinterdb_cache_flush(const splinterdb *kvs)
+{
+   cache_flush(kvs->spl->cc);
+}
+
+platform_heap_id
+splinterdb_get_heap_id(const splinterdb *kvs)
+{
+   return kvs->heap_id;
+}
+
+const void *
```

Comment:
Why are these `void *` instead of the correct type?

Reply:
These accessor methods are used in the `unit/splinterdb_forked_child_test.c` test, to validate that the sub-system's handle address is valid.

We cross-check using `platform_valid_addr_in_shm()`, which just needs a `void *` ptr of the address being validated.

If each of these `splinterdb_get...()` methods were to return the correct type, it will require further casting when calling the validation function named above.

As these interfaces are meant only for testing purposes, I felt it was acceptable just to return `void *`.

Reply:
Would you mind changing them anyway?  I feel strongly about this.

Reply:
@rtjohnso -- Changed as requested. Please see the delta-diffs in [this commit](https://github.com/vmware/splinterdb/pull/568/commits/1795b6acaa91b0841c6ab57c766d7d4a1b3bb9ef).

Curiously, I did not seem to have to change the callers to cast the return ptr to `(void *)`. Example, this one in `splinterdb_forked_child_test.c`:

```
152       ASSERT_TRUE(platform_valid_addr_in_heap(
153          spl_heap_id, splinterdb_get_task_system_handle(spl_handle)));
```

I would have expected to have to cast the return to `void *`, but I didn't get any compilation errors in my dev builds. Will see how things fare in CI.

For proper-form, I could do the casting, but seems unnecessary as compilation is not failing.
Let me know if you feel strongly that the call-sites should be amended, too.